src/main/java/org/wildfly/swarm/plugin/process/CDIMarker.java

```diff
@@ -34,7 +34,7 @@ public FractionMetadata apply(FractionMetadata meta) {
                         "</beans>");
                 writer.flush();
             } catch (IOException e) {
-                this.log.error(e.getMessage(), e);
+                throw new MojoExecutionException("Failed writing " + CDI_MARKER, e);
```

Comment:
What do we gain by being more restrictive here and in the other places where `log.error` was translated to throwing an exception?

Reply:
Failing a build when the necessary pieces of a fraction are not generated successfully. The fraction just isn't complete.